### PR TITLE
[Parser] Parse conditional expressions

### PR DESCRIPTION
**Pull request by akshanshbhatt**
Related issue: #508 
Ref: [Conditional Expressions](https://docs.python.org/3/reference/expressions.html#conditional-expressions)

### CODE REVIEW

File: src/lpython/parser/parser.yy [hunk 1]
```diff
@@ -238,14 +238,15 @@ void yyerror(YYLTYPE *yyloc, LFortran::Parser &p, const std::string &msg)
...
+%type <ast> ternary_if_statement
 
 // Precedence
 
 %precedence ":="
 %left "or"
 %left "and"
 %precedence "not"
-%left "==" "!=" ">=" ">" "<=" "<" TK_IS_NOT KW_IS TK_NOT_IN // "in"
+%left "==" "!=" ">=" ">" "<=" "<" TK_IS_NOT KW_IS TK_NOT_IN KW_IF KW_ELSE// "in"
```
Comment (Thirumalai-Shaktivel) [inline src/lpython/parser/parser.yy:249]:
```suggestion
%left "==" "!=" ">=" ">" "<=" "<" TK_IS_NOT KW_IS TK_NOT_IN // "in"
%left KW_IF KW_ELSE
```

**Comment by Thirumalai-Shaktivel** [comment]
LGTM, Thanks for this PR @akshanshbhatt!

File: src/lpython/parser/parser.yy [hunk 2]
```diff
@@ -744,6 +749,9 @@ expr
     | expr "or" expr { $$ = BOOLOP($1, Or, $3, @$); }
     | "not" expr { $$ = UNARY($2, Not, @$); }
 
+    /* | expr KW_IF expr KW_ELSE expr { } */
```
Comment (Thirumalai-Shaktivel) [inline src/lpython/parser/parser.yy:752]: Can you please remove this comment?

**Review by certik — APPROVED** [review]
@Thirumalai-Shaktivel go ahead and merge.

**Comment by Thirumalai-Shaktivel** [comment]
Thanks for the review!

**Pull request merged**